dags/hivemind_etl_helpers/src/db/discord/fetch_raw_messages.py

```diff
@@ -0,0 +1,137 @@
+                        {"isGeneratedByWebhook": False},
+                    ],
+                }
+            )
+            .sort("createdDate", 1)
+        )
+        raw_messages = list(cursor)
+    else:
+        cursor = (
+            client[guild_id]["rawinfos"]
+            .find({"isGeneratedByWebhook": False})
+            .sort("createdDate", 1)
+        )
+        raw_messages = list(cursor)
+
+    return raw_messages
+
+
+def fetch_raw_msg_grouped(
+    guild_id: str, from_date: datetime | None = None, sort: int = 1
+) -> list[dict[str, dict]]:
+    """
+    fetch raw messages grouped by day
+    this would fetch the data until 1 day ago
+
+    Parameters
+    -----------
+    guild_id : str
+        the guild id to fetch their `rawinfos` messages
+    from_date : datetime
+        get the raw data from a specific date
+        default is None, meaning get all the messages
+    sort : int
+        sort the data Ascending or Descending
+        `1` represents for Ascending
+        `-1` represents for Descending
+
+    Returns
+    --------
+    raw_messages_grouped : list[dict[str, list]]
+        ascending sorted list of raw messages
+        it would be a list, each having something like below
+        the date is in format of `%Y-%m-%d`
+        ```
+                "_id": {
+                    "date": str
+                },
+                "messages": dict[str, Any],
+        ```
+    """
+    client = MongoSingleton.get_instance().client
+
+    # the pipeline to apply through mongodb
+    pipeline: list[dict] = []
+
+    if from_date is not None:
+        pipeline.append(
+            {
+                "$match": {
+                    "$and": [
+                        {
+                            "createdDate": {
+                                "$gte": from_date,
+                                "$lt": datetime.now().replace(
+                                    hour=0, minute=0, second=0, microsecond=0
+                                ),
+                            }
+                        },
+                        {"isGeneratedByWebhook": False},
+                    ]
+                }
+            },
+        )
+    else:
+        pipeline.append(
+            {
+                "$match": {
+                    "isGeneratedByWebhook": False,
+                }
+            },
+        )
+
+    # sorting
+    pipeline.append(
+        {"$sort": {"createdDate": sort}},
+    )
+
+    # add the grouping
+    pipeline.append(
+        {
+            "$group": {
+                "_id": {
+                    "date": {
+                        "$dateToString": {"format": "%Y-%m-%d", "date": "$createdDate"}
+                    },
+                },
+                "messages": {"$push": "$$ROOT"},
+            }
+        }
+    )
+
+    cursor = client[guild_id]["rawinfos"].aggregate(pipeline)
+    raw_messages_grouped = list(cursor)
+
+    return raw_messages_grouped
```

Comment:
Just to check: The raw messages that are passed to the summarizer, are they in chronological order? So having the oldest messages first, followed by the second oldest etc.

Reply:
I'm sorting by newest first to the oldest (Ascending). The reason is in case of the script breaking in between the work it would know where to pick up the job as the newer messages were processed and it will continue from the messages after (So no it isn't the oldest first).

Was there a specific reason you mentioned the oldest first (Descending)?

Reply:
This specifically entails for the messages that are sent to the LLM: We (including LLMs) read text lines from top to bottom. In doing so, we assume that messages in the top are older than messages in the bottom. This is also how messages are shown in discord. If we reverse the order and have newest first, the LLM reads the conversation in reverse which will make it much harder to understand what was being discussed.

Reply:
I'm assuming messages at the top of a conversation is newer and as we read more we would move on to the older messages. For example if the messages were from 9:00AM to 10:00PM, we would first read the morning messages till the night messages. Am I right or I'm still missing a point here?

Reply:
I think we can't make that assumption. For example, in a discord thread, the older messages are listed above the newer messages. Or when reading a dialog in a book, the conversation has the older text above the newer text. Similarly in the prompt, the older messages should be listed above the newer messages.

Reply:
Oh, right. Thanks for re-clarifying. I think we're saying the same thing but I was wrong with words there.
Yes, the newer messages are the ones that will be given last to the LLM and the older ones will be given earlier to the LLM. Meaning for example of a day

We're giving the data of the morning first to the LLM and then we'll iterate to the night messages to do the summaries.

Reply:
Great, then we're on the same page :)